test/functional/outputs/azuremonitor/forward_to_azuremonitor_test.go

```diff
@@ -0,0 +1,194 @@
+// go:build !fluentd
+package azuremonitor
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	logging "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	"github.com/openshift/cluster-logging-operator/internal/constants"
+	"github.com/openshift/cluster-logging-operator/internal/runtime"
+	"github.com/openshift/cluster-logging-operator/internal/utils"
+	"github.com/openshift/cluster-logging-operator/test/framework/functional"
+	"github.com/openshift/cluster-logging-operator/test/helpers/azuremonitor"
+	"github.com/openshift/cluster-logging-operator/test/helpers/oc"
+	"github.com/openshift/cluster-logging-operator/test/helpers/rand"
+	"github.com/openshift/cluster-logging-operator/test/helpers/types"
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	azureSecret         = "azure-secret"
+	application         = "application"
+	serverStarted       = "Server started on port 3000"
+	transactionRecorded = "Transaction recorded"
+	failedReason        = "reason=\"Service call failed. No retries or retries exhausted.\""
+)
+
+var _ = Describe("Forwarding to Azure Monitor Log ", func() {
+	var (
+		framework  *functional.CollectorFunctionalFramework
+		mockServer *azuremonitor.MockoonServer
+		secret     *v1.Secret
+		sharedKey  = rand.Word(16)
+		inputName  = "custom-app"
+	)
+
+	BeforeEach(func() {
+		framework = functional.NewCollectorFunctionalFrameworkUsingCollector(logging.LogCollectionTypeVector)
+		mockServer = azuremonitor.NewMockoonServer(framework.Namespace, framework.Labels)
```

Comment:
Looks like we are closer and the reason I suggested to modify `/etc/hosts` if needed to make this work.  With our functional framework it is only necessary to add a container to the existing pod.  You likely will not require any of the other existing support resources other then maybe configmaps.  Additionally this structure allows you to share a disk if needed between the collector and receiver.

One should be able to add a visitor that adds in the bits needed for the container

Reply:
done